src/advanced/soft-play-advanced.js

```diff
@@ -0,0 +1,174 @@
+// do not change these lines
+// let adults = 0
+// let children = 0
+// let adultsLeaving = 0
+// let childrenLeaving = 0
+
+// TODO: Write your functions in the below section. Your functions should update
+// the adults and children variables defined above.
+// Start with the occupancy function.
+class Softplay {
+  constructor(entranceRatio, occupancyRatio) {
```

Comment:
Nice use of a class. I'd recommend these ratios also being a class so you don't have to rely on the user inputting a valid string.

```js
class Ratio {
  constructor(left, right)
}
```